tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -397,11 +413,25 @@ def media_files(self):
 
     def delete_all_sms(self):
         self.marionette.switch_to_frame()
-        return self.marionette.execute_async_script("return GaiaDataLayer.deleteAllSms();", special_powers=True)
+        self.marionette.push_permission('sms', True)
```

Comment:
Could we perhaps have a `with_permission` in Marionette that resets the permission back to the original value? I appreciate this would likely be a separate enhancement. At the moment we often assume the original value or leave the new value behind.